src/models/state/networking_state.rs

```diff
@@ -94,6 +94,12 @@ pub struct NetworkingState {
     /// record latest successful upgrade, merely latest attempt. This is to
     /// prevent excessive runs of the proof-upgrade functionality.
     pub last_tx_proof_upgrade_attempt: std::time::SystemTime,
+
+    /// Disconnection times of past peers. Can be used to determine if a connection
+    /// request should be accepted or rejected.
+    ///
+    /// Both the main task and the peer tasks may update this map.
```

Comment:
Should the main task be allowed to update this map? As far as I can tell, it only happens in `register_peer_disconnection` which, as far as I can tell, is only called from `peer_loop.rs` (and a test in `connect_to_peers.rs`).

Reply:
Does the main task only terminate connections by communicating to the peer tasks? If so, you are right, the main task should not directly change the map.

I worded it this way to not accidentally scare future us when stumbling across this.

Reply:
It looks like there is a third escape option:

 3. Trigger an error (instead of a `Ok` return code).

This escape path is taken when the peer misbehaves, resulting in a ban. And possibly other error cases also. In this case we also do not care about recording disconnection times, but even that point is moot as it all takes place inside `peer_loop.rs`.